### PR TITLE
IDEA-169757 : support charset for html report

### PR DESCRIPTION
**Pull request by lntuition**
As last step for [IDEA-169757: Code coverage report: html file encoding](https://youtrack.jetbrains.com/issue/IDEA-169757)  and https://github.com/JetBrains/coverage-report/pull/3,  https://github.com/JetBrains/intellij-coverage/pull/57, we need to modify interface to inject charset from configuration.

Reviews will be of great help to resolve issue i'm working on 😄 thanks, bro :)

### CODE REVIEW

File: src/main/kotlin/kotlinx/kover/gradle/plugin/dsl/KoverReportExtension.kt [hunk 1]
```diff
@@ -388,6 +388,13 @@ public interface KoverHtmlReportConfig {
      */
     public var title: String?
 
+    /**
+     * Specify charset in HTML reports.
+     *
+     * If not specified, Charset.defaultCharset() is used instead.
```
Comment (shanshin) [inline src/main/kotlin/kotlinx/kover/gradle/plugin/dsl/KoverReportExtension.kt:394]: UTF-8 is default for [JaCoCo](https://www.jacoco.org/jacoco/trunk/doc/ant.html)

File: src/main/kotlin/kotlinx/kover/gradle/plugin/dsl/internal/KoverReportExtension.kt [hunk 2]
```diff
@@ -94,6 +94,8 @@ internal open class KoverHtmlReportConfigImpl @Inject constructor(private val ob
 
     override var title: String? = null
 
+    override var charset: String? = null
```
Comment (shanshin) [inline src/main/kotlin/kotlinx/kover/gradle/plugin/dsl/internal/KoverReportExtension.kt:97]: Please also override this property in `KoverHtmlReportConfigWriter`

**Comment by shanshin** [comment]
Could you run `:apiDump` Gradle task to update `.api` file?

File: src/main/kotlin/kotlinx/kover/gradle/plugin/dsl/KoverReportExtension.kt (hunk 1 above)
Comment (lntuition) [reply src/main/kotlin/kotlinx/kover/gradle/plugin/dsl/KoverReportExtension.kt:394]: ok! i'll change comment 😄

File: src/main/kotlin/kotlinx/kover/gradle/plugin/dsl/internal/KoverReportExtension.kt (hunk 2 above)
Comment (lntuition) [reply src/main/kotlin/kotlinx/kover/gradle/plugin/dsl/internal/KoverReportExtension.kt:97]: oh i missed it! thank you

**Comment by lntuition** [comment]
> Could you run :apiDump Gradle task to update .api file?

thank you for reviewing 👍 i updated `.api` file

**Review by shanshin — APPROVED** [review]
Thanks for the effort!

**Comment by lntuition** [comment]
Thanks :) When are you planning the next release?

**Pull request merged**